### PR TITLE
plex-media-player: remove custom desktopItem

### LINKED ISSUE #60437: plex-media-player has 2 .desktop files
## Issue description

This is pretty confusing:

Screenshot from 2019-04-29 23-15-09

```sh
stanislas@nixpsla ~> find /nix/store/d50hm4vhyw0nib9zflf6nqsqyw5l7rql-plex-media-player-2.23.0.920/ -name "*.desktop"
/nix/store/d50hm4vhyw0nib9zflf6nqsqyw5l7rql-plex-media-player-2.23.0.920/share/applications/plex-media-player.desktop
/nix/store/d50hm4vhyw0nib9zflf6nqsqyw5l7rql-plex-media-player-2.23.0.920/share/applications/plexmediaplayer.desktop
```

```sh
stanislas@nixpsla ~> cat /nix/store/d50hm4vhyw0nib9zflf6nqsqyw5l7rql-plex-media-player-2.23.0.920/share/applications/plex-media-player.desktop
[Desktop Entry]
Type=Application
Exec=plexmediaplayer
Terminal=false
Name=Plex Media Player
Categories=AudioVideo;Video;Player;TV;
Icon=plex-media-player
Comment=View your media
GenericName=Media Player
```

```sh
stanislas@nixpsla ~> cat /nix/store/d50hm4vhyw0nib9zflf6nqsqyw5l7rql-plex-media-player-2.23.0.920/share/applications/plexmediaplayer.desktop
[Desktop Entry]
Version=1.0
Name=Plex Media Player
GenericName=Media Player
Comment=View your media
Exec=plexmediaplayer --fullscreen --tv
Icon=plexmediaplayer
Terminal=false
Type=Application
Categories=AudioVideo;Video;Player;TV;

Actions=TV;DesktopF;DesktopW;

[Desktop Action TV]
Name=TV
Exec=plexmediaplayer --fullscreen --tv

[Desktop Action DesktopF]
Name=Desktop [Fullscreen]
Exec=plexmediaplayer --fullscreen --desktop

[Desktop Action DesktopW]
Name=Desktop [Windowed]
Exec=plexmediaplayer --windowed --desktop
```

### Steps to reproduce

Install `plex-media-player` and try to launch it from your DE. Which icon do you click on? They both do the same thing anyway.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 5.0.9, NixOS, 19.09pre177249.dfd8f84aef1 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"nixos-19.09pre177249.dfd8f84aef1, nixos-hardware"`
 - channels(stanislas): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by angristan**
###### Motivation for this change

Fixes #60437: `plex-media-player` comes with a working desktop file, so there is no need to add our own custom one unless we remove the original. Both seem to work fine, so let's keep it simple.

###### Things done

- Removed `makeDesktopItem` dependency
- Removed creation and copy of the custom desktop item `desktopItem` (`plex-media-player.desktop`)

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by alexarice** [comment]
I have tested this and can confirm it builds and had two desktop items before and now has 1. I have not tested any binaries

File: pkgs/applications/video/plex-media-player/default.nix [hunk 1]
```diff
@@ -72,7 +62,6 @@ in stdenv.mkDerivation rec {
   postInstall = ''
     mkdir -p $out/share/{applications,pixmaps}
```
Comment (JohnAZoidberg) [inline pkgs/applications/video/plex-media-player/default.nix]: Then you don't need to create the `application` directory anymore.
You can also combine the remaining two lines with

```
install -D ${src}/resources/images/icon.png $out/share/pixmaps/plex-media-player.png
```

**Comment by JohnAZoidberg** [comment]
Great! The upstream desktop file is more fully featured anyways :)

File: pkgs/applications/video/plex-media-player/default.nix (hunk 1 above)
Comment (angristan) [reply pkgs/applications/video/plex-media-player/default.nix]: Thanks!

**Comment by jtojnar** [comment]
Is installing the icon necessary? I would expect them to install it on their own when they added a desktop file. Are the paths in the desktop file correct? Sometimes project hardcode `/usr/share` there.

**Comment by angristan** [comment]
Indeed, the install part is not needed, the icon is fine even after removing that part.

There is no path in the desktop file:

```desktop
stanislas@nixpsla ~/nixpkgs> cat result/share/applications/plexmediaplayer.desktop
[Desktop Entry]
Version=1.0
Name=Plex Media Player
GenericName=Media Player
Comment=View your media
Exec=plexmediaplayer --fullscreen --tv
Icon=plexmediaplayer
Terminal=false
Type=Application
Categories=AudioVideo;Video;Player;TV;

Actions=TV;DesktopF;DesktopW;

[Desktop Action TV]
Name=TV
Exec=plexmediaplayer --fullscreen --tv

[Desktop Action DesktopF]
Name=Desktop [Fullscreen]
Exec=plexmediaplayer --fullscreen --desktop

[Desktop Action DesktopW]
Name=Desktop [Windowed]
Exec=plexmediaplayer --windowed --desktop
```

**Review by worldofpeace — APPROVED** [review]
Tested the desktop item and all icons and actions work.

Please squash your commits accordingly and we should be good to merge :smile: 
##### one to fix `desktopItem`, another to not copy icon (or whatever)

**Comment by angristan** [comment]
I should be good now, I hope :slightly_smiling_face:

**Comment by worldofpeace** [comment]
:sparkles: Thank you @angristan for contributing

**Comment by angristan** [comment]
Thanks everyone for helping out! :bowing_man:

**Pull request merged**